sequence_processing_pipeline/NuQCJob.py

```diff
@@ -261,40 +271,15 @@ def run(self, callback=None):
                                            empty_files_directory,
                                            self.minimum_bytes)
 
-    def _get_failed_indexes(self, project_name, job_id):
-        pattern = f"{self.temp_dir}/{self.batch_prefix}.*.completed"
+    def _confirm_job_completed(self):
+        # since NuQCJob processes across all projects in a run, there isn't
+        # a need to iterate by project_name and job_id.
+        pattern = f"{self.output_path}/hds-{self.qiita_job_id}.*.completed"
         completed_files = list(glob.glob(pattern))
-        completed_indexes = []
-        regex = r'^%s/%s.%s_([0-9]+).completed$' % (self.temp_dir,
-                                                    self.batch_prefix,
-                                                    str(job_id))
-        array_ids = re.compile(regex)
-
-        for completed_file in completed_files:
-            match = array_ids.search(completed_file)
-            if match is None:
-                raise PipelineError("Malformed completed file")
-            else:
-                id_ = int(match.groups(0)[0])
-                completed_indexes.append(id_)
-
-        # a successfully completed job array should have a list of array
-        # numbers from 0 - len(self.commands).
-        all_indexes = list(range(1, self.counts[self.batch_prefix]))
-        failed_indexes = sorted(set(all_indexes) - set(completed_indexes))
-
-        # generate log-file here instead of in run() where it can be
-        # unittested more easily.
-        log_fp = join(self.output_path,
-                      'logs',
-                      f'failed_indexes_{job_id}.json')
-
-        if failed_indexes:
-            with open(log_fp, 'w') as f:
-                f.write(dumps({'job_id': job_id,
-                               'failed_indexes': failed_indexes}, indent=2))
```

Comment:
Just to confirm that we do _not_ need this validation any longer ...

Reply:
yup! the new code generates a single .completed file rather than individual .completed files for each sample.